drivers/frequency/adf4377/adf4377.c

```diff
@@ -0,0 +1,669 @@
+	if (ret < 0)
+		return ret;
+
+	ret = adf4377_spi_read(dev, ADF4377_REG(0x0A), &scratchpad);
+	if (ret < 0)
+		return ret;
+
+	if(scratchpad != ADF4377_SPI_SCRATCHPAD_B)
+		return FAILURE;
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Set default datasheet register values.
+ * @param dev - The device structure.
+ * @return Returns SUCCESS in case of success or negative error code.
+ */
+static int32_t adf4377_set_default(struct adf4377_dev *dev)
+{
+	int32_t ret;
+
+	ret = adf4377_spi_write(dev, ADF4377_REG(0x0f), ADF4377_R00F_RSV1);
+	if (ret < 0)
+		return ret;
+
+	ret = adf4377_update(dev, ADF4377_REG(0x1c), ADF4377_R01C_RSV1_MSK,
+			     ADF4377_R01C_RSV1(0x1));
+	if (ret < 0)
+		return ret;
+
+	ret = adf4377_update(dev, ADF4377_REG(0x1f), ADF4377_R01F_RSV1_MSK,
+			     ADF4377_R01F_RSV1(0x7));
+	if (ret < 0)
+		return ret;
+
+	ret = adf4377_update(dev, ADF4377_REG(0x20), ADF4377_R020_RSV1_MSK,
+			     ADF4377_R020_RSV1(0x1));
+	if (ret < 0)
+		return ret;
+
+	ret = adf4377_spi_write(dev, ADF4377_REG(0x21), ADF4377_R021_RSV1);
+	if (ret < 0)
+		return ret;
+
+	ret = adf4377_spi_write(dev, ADF4377_REG(0x22), ADF4377_R022_RSV1);
+	if (ret < 0)
+		return ret;
+
+	ret = adf4377_spi_write(dev, ADF4377_REG(0x23), ADF4377_R023_RSV1);
+	if (ret < 0)
+		return ret;
+
+	ret = adf4377_update(dev, ADF4377_REG(0x25), ADF4377_R025_RSV1_MSK,
+			     ADF4377_R025_RSV1(0xB));
+	if (ret < 0)
+		return ret;
+
+	ret = adf4377_spi_write(dev, ADF4377_REG(0x2C), ADF4377_R02C_RSV1);
+	if (ret < 0)
+		return ret;
+
+	ret = adf4377_spi_write(dev, ADF4377_REG(0x31), ADF4377_R031_RSV1);
+	if (ret < 0)
+		return ret;
+
+	ret = adf4377_update(dev, ADF4377_REG(0x32), ADF4377_R032_RSV1_MSK,
+			     ADF4377_R032_RSV1(0x9));
+	if (ret < 0)
+		return ret;
+
+	ret = adf4377_spi_write(dev, ADF4377_REG(0x33), ADF4377_R033_RSV1);
+	if (ret < 0)
+		return ret;
+
+	ret = adf4377_spi_write(dev, ADF4377_REG(0x34), ADF4377_R034_RSV1);
+	if (ret < 0)
+		return ret;
+
+	ret = adf4377_spi_write(dev, ADF4377_REG(0x3A), ADF4377_R03A_RSV1);
+	if (ret < 0)
+		return ret;
+
+	ret = adf4377_spi_write(dev, ADF4377_REG(0x3B), ADF4377_R03B_RSV1);
+	if (ret < 0)
+		return ret;
+
+	return adf4377_spi_write(dev, ADF4377_REG(0x42), ADF4377_R042_RSV1);
+}
+
+/**
+ * @brief Software reset.
+ * @param dev - The device structure.
+ * @return Returns SUCCESS in case of success or negative error code.
+ */
+int32_t adf4377_soft_reset(struct adf4377_dev *dev)
+{
+	int32_t ret;
+	uint16_t timeout = 0xFFFF;
+	uint8_t data;
+
+	ret = adf4377_update(dev, ADF4377_REG(0x00),
+			     ADF4377_SOFT_RESET_MSK | ADF4377_SOFT_RESET_R_MSK,
+			     ADF4377_SOFT_RESET(ADF4377_SOFT_RESET_EN) | ADF4377_SOFT_RESET_R(
+				     ADF4377_SOFT_RESET_EN));
+	if (ret < 0)
+		return ret;
+
+	while(timeout--) {
+		ret = adf4377_spi_read(dev, ADF4377_REG(0x00), &data);
+		if (ret < 0)
+			return ret;
+
+		if(!(data & ADF4377_SOFT_RESET(ADF4377_SOFT_RESET_EN)))
+			return adf4377_set_default(dev);
```

Comment:
Doesn't the software reset update already to the default values?

Reply:
Nope, the default values must be rewritten at each re-initialization of the device. I know it's a bit unusual, but this is how this part works.